server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/executeBash.ts

```diff
@@ -328,11 +328,28 @@ export class ExecuteBash {
 
             this.childProcess = new ChildProcess(this.logging, 'bash', ['-c', params.command], childProcessOptions)
 
-            // Set up cancellation listener
+            // Setup a periodic check for trigger cancellation
             if (cancellationToken) {
                 cancellationToken.onCancellationRequested(() => {
-                    this.logging.debug('Cancellation requested, killing child process')
-                    this.childProcess?.stop()
+                    if (cancellationToken.isCancellationRequested) {
```

Comment:
this line is redundant, the token cancellation is already requested at this point